lib/jsonapi/plugs/underscore_parameters.ex

```diff
@@ -55,15 +67,38 @@ defmodule JSONAPI.UnderscoreParameters do
   alias JSONAPI.Utils.String, as: JString
 
   @doc false
-  def init(opts), do: opts
+  def init(opts) do
```

Comment:
The following is certainly functional and raising an error for the wrong type is really nice; I'd like to suggest a stylistic change if you don't mind: Using two function clauses, one that matches when `replace_query_params` is not a boolean (that clause can simply raise your error) and then a default clause with the same definition as `init/1` had before to just pass `opts` on in all other cases.

I can say more if you're not sure what I am picturing.

Reply:
Roger that.  I will make the change now.

Reply:
Alternatively we could remove the checking in the `init` block.  I don't think it is 100% necessary, but it is a nice-to-have.

Reply:
Hmmm, well, I was conflating Elixir's behavior when pattern matching dictionaries with its behavior when pattern matching lists so the solution I pictured is not valid after all.

> because if the opts list ever (accidentally) contained more than one element, the pattern wouldn't match and our check would be skipped...

We could also check the length of the list to throw an error about only supporting one option, but that's not a very sustainable solution either because the second we do add a second option we need to re-evaluate our pattern matching strategy anyway.

Thanks for giving it a try. I think your original code here feels best given the error you are raising is a very nice developer experience.